metric-schema-java/src/main/java/com/palantir/metric/schema/UtilityGenerator.java

```diff
@@ -208,22 +250,27 @@ private static void generateMetricFactoryBuilder(
                 .addModifiers(visibility.apply())
                 .addMethods(abstractBuildMethods)
                 .build());
-        ImmutableList<String> tagList = ImmutableList.copyOf(definition.getTags());
+        ImmutableList<TagDefinition> tagList = ImmutableList.copyOf(definition.getTagDefinitions());
```

Comment:
@carterkozak it seems like we had previously relied on the implicit order of the set when generating the builder stages. I'll make it alphabetical based off of the tag name, but this may some impact consumers

Reply:
We discussed tag order when we initially implemented metric-schema :-)
Unlike conjure, we want to reflect the input because it may be meaningful, and changes only impact code in the consumer project. Stable order doesn't buy us anything and costs legibility.

Take our metrics with `service` and `endpoint` tags for example, you'd expect to start with `service` and build specificity toward the `endpoint` tag. Similarly, if you have a series of metrics with both `service` and `endpoint` tags, you may have additional tags on some metrics, e.g. `framework` or `channel-name`. imo it's jarring if these break a common pattern.

Reply:
Ok, I've updated the api and made it explicit that we will maintain the order of tags.